system_setup/server/controllers/configure.py

```diff
@@ -37,6 +40,90 @@ def __init__(self, app):
     def start(self):
         self.install_task = self.app.aio_loop.create_task(self.configure())
 
+    def _update_locale_gen(self, localeGenPath, lang):
+        """ Uncomments the line in locale.gen file where lang is found,
+            if found commented. A fully qualified language is expected,
+            since that would have passed thru the Locale
+            controller validation. e.g. en_UK.UTF-8. """
+
+        fileContents: str
+        locGenNeedsWrite = False
+        with open(localeGenPath, "r") as f:
+            fileContents = f.read()
+            lineFound = fileContents.find(lang)
+            if lineFound == -1:
+                # An unsupported locale coming from our UI is a bug
+                log.error("Selected language %s not supported.", lang)
+                return
+
+            pattern = r'#+\s*({}.*)'.format(lang)
+            commented = re.compile(pattern)
+            (fileContents, n) = commented.subn(r'\1', fileContents, count=1)
+            locGenNeedsWrite = n == 1
+
+        if locGenNeedsWrite:
```

Comment:
you can early return reversing the condition :)

Reply:
Redesigned those functions to enable early returns across the board.